articles/grand-rename.md

```diff
@@ -0,0 +1,97 @@
+
+After more than 3 years working on TelemetryDeck, we think it's about time to streamline things for a more clear long-term future.
+
+## Consistency, Longevity, and Clarity
+
+Our main goals with our new naming scheme were to be internally consistent, but at the same time make sure the names are abstract enough to work flexibly for changes to come in the foreseeable future. Too abstract names can be confusing and cumbersome, so we additionally strived for clarity.
+
+The most obvious change was to rename our Swift repository from "SwiftClient" to "SwiftSDK". Thanks to GitHub redirections, this change should have no effect on our existing customers, but long term, it should make things more consistent. 🎉
+
+For our signals & parameters, we decided the prefix them with `TelemetryDeck` to make the source clear and avoid amgiguity. Additionally, we analyzed all the signals and parameters we currently send and future ones we have in our roadmap and grouped them as follows:
+
+### Signals
+
+- `Session`: Anything related to starting, pausing, continuing, or ending a user session.
+
+### Parameters
+
+- `AppContext`: The context the app runs in, such as the locale, the debug mode, or the target environment.
+- `AppInfo`: Information about the specific app build, such as version, build number, or SDKs compiled with.
+- `Device`: All about the device running the application, such as operating system, model name, or architecture.
+- `SDK`: Information about the TelemetryDeck SDK, such as the version number or supported features.
+
+{% noteinfo "No action needed" %}
+To make sure these changes don't affect existing customers with existing insights anytime soon, we decided to keep sending the old names for a transition period of at least 1 year while also sending the new ones. We have also auto-migrated any insights that were using the old names in their filters to work with the new system. This was safely achieved by replacing filters like "platform == iOS" with something like "platform == iOS OR TelemetryDeck.SystemInfo.platform == iOS" accepting both styles, ensuring you will continue to see historic data from before the transition, during the transition, and after the transition.
+{% endnoteinfo %}
+
+<!-- Discussion: What about Top-N insights that use a parameter like "appVersion" which got renamed to "TelemetryDeck.SystemInfo.appVersion"? -->
```

Comment:
I think those by sending both new and deprecated parameter names in parallel for a good long while, we can then auto-update those in the future. (see also previous comment)

Reply:
Makes sense!

Reply:
@kkostov:
> I can imagine some folks may ask if sending both signal types would also mean they'd be billed as two signals?

Unless I'm misunderstanding @Jeehut, we're only sending duplicate parameters in the same signal. Those don't cost extra.

(I'll try to reply to the rest tomorrow)

Reply:
Yes, that's what I meant. I do realize though that he has a point that we also have a signal that we're renaming. But it's just one signal, so I'm not sure if it's a big deal for customers. If anything, I would suggest that we allow them to fully migrate to the new system and avoid duplicate calls by skipping the deprecated call. But in any case I would keep the duplicate parameters for both signals.

Reply:
I removed this comment in 8756fd592da067913ea5cbb96b61ce3670448951 and I agree with you that because we're sending the old names in parallel rather than doing the auto-migration, this is a non-issue at this moment.